File: kotlin/3_operator/unary/Decrement.kt
fun main() {
    
    var a: Int = 5
    var hasil: Int? = null
    
    hasil = a--
    println("Hasil variable a=5 di decrement a-- adalah: $hasil")
    a = 5
    hasil = --a
    println("Hasil variable a=5 di decrement --a adalah: $hasil")
}

File: c++/3_operator/5_unary/Decrement.cpp
#include <iostream>
using namespace std;

int main(){

    int a = 5;
    int hasil;

    hasil = a--;
    cout << "Hasil variable a=5 decrement a--, simpan ke hasil: " << hasil << " nilai a: " << a << endl;

    a = 5;
    hasil = --a;
    cout << "Hasil variable a=5 decrement --a, simpan ke hasil: " << hasil << " nilai a: " << a << endl;

    cin.get();
    return 0;
}

/*
Pada operator decrement setiap kali a-- dieksekusi itu sama saja dengan mengurangi satu -1 nilai di variabel a
terdapat 2 cara yaitu:
a-- post-increment
--a pre-decrement
*/